### PR TITLE
ODIM-5648: Added Aggregation action info under AggregationService 

### PR DESCRIPTION
**Pull request by chauberahul1993**
ODIM-5648: Added Aggregation action info under AggregationService

### CODE REVIEW

File: svc-aggregation/rpc/aggregator.go [hunk 1]
```diff
@@ -894,3 +897,85 @@ func (a *Aggregator) SendStartUpData(ctx context.Context, req *aggregatorproto.S
...
+				Name:            "ResetType",
+				Required:        true,
+				DataType:        "string",
+				AllowableValues: []string{"On", "ForceOff", "GracefulShutdown", "GracefulRestart", "ForceRestart", "Nmi", "ForceOn", "PushPowerButton"},
+			},
+			{
+				Name:     "TargetURIs",
+				Required: true,
+				DataType: "array",
+				// AllowableValues: []string{"On", "ForceOff", "GracefulShutdown", "GracefulRestart", "ForceRestart", "Nmi", "ForceOn", "PushPowerButton"},
```
Comment (irappa-pattar) [inline svc-aggregation/rpc/aggregator.go:934]: remove commented lines

File: svc-aggregation/rpc/aggregator.go [hunk 2]
```diff
@@ -894,3 +897,85 @@ func (a *Aggregator) SendStartUpData(ctx context.Context, req *aggregatorproto.S
...
+		Name:      "Reset Action Info",
+		Parameters: []agresponse.Parameter{
+			{
+				Name:            "ResetType",
+				Required:        true,
+				DataType:        "string",
+				AllowableValues: []string{"On", "ForceOff", "GracefulShutdown", "GracefulRestart", "ForceRestart", "Nmi", "ForceOn", "PushPowerButton"},
+			},
+			{
+				Name:     "TargetURIs",
```
Comment (irappa-pattar) [inline svc-aggregation/rpc/aggregator.go:931]: there are other two optional parameters are there, add those also. 
BatchSize and DelayBetweenBatchesInSeconds

File: svc-aggregation/rpc/aggregator.go [hunk 3]
```diff
@@ -894,3 +897,85 @@ func (a *Aggregator) SendStartUpData(ctx context.Context, req *aggregatorproto.S
...
+	setDefaultBootOrderActionInfoResponse, _ := json.Marshal(agresponse.ActionInfo{
+		ID:        "SetDefaultBootOrderActionInfo",
+		OdataType: "#ActionInfo.v1_2_0.ActionInfo",
+		Name:      "SetDefaultBootOrder Action Info",
+		Parameters: []agresponse.Parameter{
+			{
+				Name:     "Systems",
+				Required: true,
+				DataType: "Array",
+				// AllowableValues: []string{"On", "ForceOff", "GracefulShutdown", "GracefulRestart", "ForceRestart", "Nmi", "ForceOn", "PushPowerButton"},
```
Comment (irappa-pattar) [inline svc-aggregation/rpc/aggregator.go:972]: remove commented lines

File: svc-api/handle/aggregator.go [hunk 4]
```diff
@@ -788,3 +790,67 @@ func (a *AggregatorRPCs) GetConnectionMethod(ctx iris.Context) {
...
+		errorMessage := "no X-Auth-Token found in request header"
+		response := common.GeneralError(http.StatusUnauthorized, response.NoValidSession, errorMessage, nil, nil)
+		common.SetResponseHeader(ctx, response.Header)
+		ctx.StatusCode(http.StatusUnauthorized)
+		ctx.JSON(&response.Body)
+		return
+	}
+	resp, err := a.GetResetActionInfoServiceRPC(req)
+	if err != nil {
+		errorMessage := "something went wrong with the RPC calls: " + err.Error()
```
Comment (irappa-pattar) [inline svc-api/handle/aggregator.go:810]: give some proper error message, like ("RPC call error: " + err.Error())

File: svc-api/handle/aggregator.go [hunk 5]
```diff
@@ -788,3 +790,67 @@ func (a *AggregatorRPCs) GetConnectionMethod(ctx iris.Context) {
...
+		errorMessage := "no X-Auth-Token found in request header"
+		response := common.GeneralError(http.StatusUnauthorized, response.NoValidSession, errorMessage, nil, nil)
+		common.SetResponseHeader(ctx, response.Header)
+		ctx.StatusCode(http.StatusUnauthorized)
+		ctx.JSON(&response.Body)
+		return
+	}
+	resp, err := a.GetSetDefaultBootOrderActionInfoRPC(req)
+	if err != nil {
+		errorMessage := "something went wrong with the RPC calls: " + err.Error()
```
Comment (irappa-pattar) [inline svc-api/handle/aggregator.go:842]: same as above comment

**Review by irappa-pattar — APPROVED** [review]

File: svc-api/handle/aggregator.go (hunk 5 above)
Comment (irappa-pattar) [inline svc-api/handle/aggregator.go:842]: same as above comment

**Review by irappa-pattar — CHANGES_REQUESTED** [review]

File: svc-aggregation/rpc/aggregator.go (hunk 3 above)
Comment (chauberahul1993) [reply svc-aggregation/rpc/aggregator.go]: done

File: svc-aggregation/rpc/aggregator.go (hunk 2 above)
Comment (chauberahul1993) [reply svc-aggregation/rpc/aggregator.go:930]: fixed

File: svc-api/handle/aggregator.go (hunk 4 above)
Comment (chauberahul1993) [reply svc-api/handle/aggregator.go]: done

File: svc-api/handle/aggregator.go (hunk 5 above)
Comment (chauberahul1993) [reply svc-api/handle/aggregator.go]: done

File: svc-aggregation/rpc/aggregator.go (hunk 1 above)
Comment (chauberahul1993) [reply svc-aggregation/rpc/aggregator.go]: done

**Review by irappa-pattar — APPROVED** [review]

**Review by amar-shalgar — APPROVED** [review]
LGTM

**Review by jeevan-kamkar — APPROVED** [review]

**Review by hemkapoor — APPROVED** [review]

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**